### DISCUSSION TITLE
How to make a task always be scheduled even previous task failed?

### DISCUSSION THREAD

**Discussion by hankquan** [Netflix/conductor#2700] [Q&A]
Is there possible to make a task always be scheduled even previous task failed? Just like a finally task, the purpose of this requirement is that I would like to do some cleaning actions even the workflow failed somewhere. Thanks

**Comment by maheshyaddanapudi** [comment]
I think if you set the parameter "optional": true in the previous task, it would not fail the workflow and carry forward with the next task.

**Reply by hankquan** [reply]
Thanks @maheshyaddanapudi , I didn't know this options. It helps a lot, thank you.

**Comment by apanicker-nflx** [comment]
Thanks for the answer @maheshyaddanapudi .
While setting `optional:true` on the task would prevent it from failing, it would not help achieve the cleanup action that is desired.

@howaric You would probably benefit from using the `failure_workflow` [feature](https://netflix.github.io/conductor/configuration/workflowdef/#workflow-definition) . You could create a workflow which does cleanup and link to it from your workflow definition.

**Reply by hankquan** [reply]
Thanks @apanicker-nflx , it could work I think, and I'll try.
While I think it is a little heavy, maybe I just want to do some small actions, but I need to define a workflow for it. Maybe we can implement a  system task, eg: Final Task?

**Reply by apanicker-nflx** [reply]
Defining this as a workflow makes this more composable and reusable. Please take a look at #2705 where the idea is to publish an event into an internal queue for every workflow, which can be extended to solve for both the "exception" clause and the "finally" clause.
